apps/client/components/ShopifyPurchaseOrdersTable.tsx

```diff
@@ -0,0 +1,266 @@
+'use client';
+import { PurchaseOrderLineItems, PurchaseOrders } from '@prisma/client';
+import {
+  IndexTable,
+  Card,
+  useIndexResourceState,
+  Text,
+  useBreakpoints,
+} from '@shopify/polaris';
+import { IndexTableSortDirection } from '@shopify/polaris/build/ts/src/components/IndexTable';
+
+import React, { useMemo, useState } from 'react';
+
+export type PurchaseOrdersRow = PurchaseOrders & {
+  purchase_order_line_items: PurchaseOrderLineItems[];
+};
+
+type ComputedPurchaseOrdersRow = Omit<PurchaseOrdersRow, 'id'> & {
+  id: string;
+  quantity: number;
+  totalCost: number;
+};
+
+const dateFormat = (value: Date | null) => {
+  if (value === null) {
+    return '';
+  }
+  return new Date(value).toLocaleDateString();
+};
+
+const currencyFormat = (value: number) => {
+  return value.toLocaleString('en-US', {
+    style: 'currency',
+    currency: 'USD',
+  });
+};
+
+export default function ShopifyPurchaseOrdersTable({
+  purchaseOrders,
+}: {
+  purchaseOrders: PurchaseOrdersRow[];
+}) {
+  const [computedPurchaseOrdersRows, setComputedPurchaseOrdersRows] = useState<
+    ComputedPurchaseOrdersRow[]
+  >(
+    purchaseOrders.map((purchaseOrder) => {
+      const { totalCost, quantity } =
+        purchaseOrder.purchase_order_line_items.reduce(
+          (acc, item) => {
+            return {
+              quantity: acc.quantity + item.quantity,
+              totalCost: acc.totalCost + Number(item.unit_cost) * item.quantity,
+            };
+          },
+          { quantity: 0, totalCost: 0 }
+        );
+      return {
+        ...purchaseOrder,
+        id: purchaseOrder.id.toString(),
+        totalCost,
+        quantity,
+      };
+    })
+  );
+
+  const resourceName = {
+    singular: 'order',
+    plural: 'orders',
+  };
+
+  const [sortDirection, setSortDirection] = useState<IndexTableSortDirection>();
+  const [sortedColumnIndex, setSortedColumnIndex] = useState<number>();
+
+  const handleSort = (
```

Comment:
Did you consider handling sorting on the server using Prisma?  How would this work with larger datasets?

Reply:
If i was going to be working with more data I would likely move this serverside with a couple enahncements:
- pagination
- query params
- sort params
Then I could pass my current page, any sorting/query options i've done and update my state based on the response from the new query. tradeoffs here would be drastically increasing server calls at the benefit of improving page performance